components/Label/Label.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useContext } from "react"
+import styled from "styled-components"
+import { ThemeContext } from "styled-components"
+import { parseCustomColor, parseColorPreset } from "../_utils"
+import Icon from "../Icon"
+import BaseComponent from "../BaseComponent"
+import { ILabel } from "./types"
+
+const DEFAULT_COLOR_THEME = "neutral"
+
+const Label: ILabel = ({
+  expanded = false,
+  text,
+  size = "medium",
+  rounded = false,
+  outlined = false,
+  colorPreset = DEFAULT_COLOR_THEME,
+  customColor,
+  icons,
+  focus,
+  className,
+  style,
+}) => {
+  const themeContext = useContext(ThemeContext)
+  const { shades, typography } = themeContext
+  const typographySize = {
+    small: typography[12],
+    medium: typography[14],
+    large: typography[14],
+  }
+  const fontSize = typographySize[size].fontSize
+  const lineHeight = typographySize[size].lineHeight
+  const parsedColorTheme = customColor
+    ? parseCustomColor(customColor)
+    : parseColorPreset(colorPreset)
+  const backgroundColor = parsedColorTheme.background
+  const fontColor = parsedColorTheme.font
+  const darkenedBorderColor = backgroundColor.set("hsl.l", "-0.2")
+  const showLeftIcon = icons && icons.left
+  const showRightIcon = icons && icons.right
+
+  const verticalPadding = {
```

Comment:
can we move all these functions outside render

Reply:
removed